### PR TITLE
8273513: Make java.io.FilterInputStream specification more precise about overrides

### PR DESCRIPTION
**Pull request by bplb**
Modify the class level specification of `java.io.FilterInputStream` to be more exact about `java.io.InputStream` methods that it overrides.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273513](https://bugs.openjdk.java.net/browse/JDK-8273513): Make java.io.FilterInputStream specification more precise about overrides

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**) ⚠️ Review applies to 07dbe2fc457ea9d89c8bf3ee8e0cb00581f37b5b
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to 96f5ffc35dbbac9dee1295cd845b821d67d9af3a

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5426/head:pull/5426` \
`$ git checkout pull/5426`

Update a local copy of the PR: \
`$ git checkout pull/5426` \
`$ git pull https://git.openjdk.java.net/jdk pull/5426/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5426`

View PR using the GUI difftool: \
`$ git pr show -t 5426`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5426.diff">https://git.openjdk.java.net/jdk/pull/5426.diff</a>

</details>

### CODE REVIEW

**Comment by bplb** [comment]
Some other incidental modifications are made in passing, principally adding `@Override` annotations and putting statements like `This method does X` in `@implSpec` blocks.

**Comment by bplb** [comment]
CSR filed: [JDK-8273517](https://bugs.openjdk.java.net/browse/JDK-8273517).

**Review by dfuch — APPROVED** [review]
LGTM. The use of `synchronized` on `mark()` and `reset()` is strange. But I guess that should be for another day...

File: src/java.base/share/classes/java/io/FilterInputStream.java [hunk 1]
```diff
@@ -86,7 +83,8 @@ public int read() throws IOException {
      * Reads up to {@code b.length} bytes of data from this
      * input stream into an array of bytes. This method blocks until some
      * input is available.
-     * <p>
+     *
+     * @implSpec
      * This method simply performs the call
      * {@code read(b, 0, b.length)} and returns
      * the  result. It is important that it does
```
Comment (naotoj) [inline src/java.base/share/classes/java/io/FilterInputStream.java:90]: Preexisting: an extra space before "result."

**Review by naotoj — APPROVED** [review]
LGTM.

**Pull request closed without merging**